src/compiler/types.ts

```diff
@@ -1832,17 +1837,22 @@ namespace ts {
         _statementBrand: any;
     }
 
+    /** Type for statements that appear only in statement position. */
+    export interface StatementOnly extends Statement {

```

Comment:
Can we split off MissingDeclaration instead of having to create this type?

Reply:
Turns out MissingDeclaration has some supertypes it doesn't need. See #23485.